File: python/tests/transport/neopixel_test.py
import _testconfig as cfg
from periph.transport.neopixel_micropython import NeoPixelTransport

from machine import SPI, Pin

passed = 0
failed = 0


def check_true(label, condition):
    global passed, failed
    if condition:
        print('PASS', label)
        passed += 1
    else:
        print('FAIL', label)
        failed += 1


def check_eq(label, got, expected):
    global passed, failed
    if got == expected:
        print('PASS', label)
        passed += 1
    else:
        print('FAIL {}: got {}, expected {}'.format(label, got, expected))
        failed += 1


spi = SPI(cfg.SPI_BUS, baudrate=2_400_000, polarity=0, phase=0,
          sck=Pin(cfg.SCK), mosi=Pin(cfg.MOSI), miso=Pin(cfg.MISO))
transport = NeoPixelTransport(spi)

data = bytes([0xFF, 0x00, 0x00])
transport.write(data)

check_true('write accepted data', len(data) == 3)

print('===DONE: {} passed, {} failed==='.format(passed, failed))

File: nodejs/tests/transport/neopixel_test.js
'use strict';

const { NeoPixelTransport } = require('../../packages/periph/src/transport/neopixel');

const SPI_BUS    = parseInt(process.env.SPI_BUS     || '0', 10);
const SPI_DEVICE = parseInt(process.env.SPI_DEVICE  || '0', 10);

let passed = 0;
let failed = 0;

function checkTrue(label, condition) {
    if (condition) { console.log('PASS', label); passed++; }
    else           { console.log('FAIL', label); failed++; }
}

const transport = new NeoPixelTransport(SPI_BUS, SPI_DEVICE);
const data = Buffer.from([0xFF, 0x00, 0x00]);
transport.write(data);

checkTrue('write accepted data', data.length === 3);

transport.close();

console.log(`===DONE: ${passed} passed, ${failed} failed===`);
process.exit(failed === 0 ? 0 : 1);